src/Adapter/Local.php

```diff
@@ -122,10 +136,26 @@ public function includeFile($path, $once = true)
 
         try {
             return Thrower::call(__NAMESPACE__ . '\includeFile' . ($once ? 'Once' : ''), $location);
-        } catch (\Exception $e) {
+        } catch (\ErrorException $e) {
             throw new IncludeFileException($e->getMessage(), $path, 0, $e);
         }
     }
+    
+    /**
+     * @inheritdoc
+     */
+    public function getMetadata($path)
+    {
+        $location = $this->applyPathPrefix($path);
+
+        if (!file_exists($location)) {
+            throw new FileNotFoundException($path);
+        }
+
+        $info = new SplFileInfo($location);
+
+        return $this->normalizeFileInfo($info);
+    }
```

Comment:
What's the change here? Adding the `file_exists` check? The filesystem implementation should check for the file existing, checking again here shouldn't make a difference. Most of the native filesystem calls are cached anyways, so it could return true even after the file is deleted. I think we need to follow the same philosophy of the change for delete(). I'm not sure at what point the php warning/error would be emitted, either at the construction or when accessing properties. I know this sounds like a pain...what are your thoughts?

Reply:
It's just another place which sometimes breaks. I don't remember what was the exact error message, but probably something like 'stat failed' RuntimeException